src/modules/commander/Commander.cpp

```diff
@@ -1620,8 +1637,10 @@ Commander::run()
 			if (is_vtol(&status)) {
 
 				// Check if there has been any change while updating the flags
-				if (status.is_rotary_wing != vtol_status.vtol_in_rw_mode) {
-					status.is_rotary_wing = vtol_status.vtol_in_rw_mode;
+				if ((status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) != vtol_status.vtol_in_rw_mode) {
```

Comment:
Can it be that you misunderstood this? I think  this check is just to check if the variable changed and not about the actual value.